rfhub2/cli/statistics_importer.py

```diff
@@ -45,27 +45,35 @@ def get_execution_files_paths(self) -> Set[Path]:
             if self._is_valid_execution_file(p)
         }
 
-    def add_statistics(self, statistics: List[KeywordStatistics]) -> Tuple[int, int]:
+    def add_statistics(
+        self, statistics: List[KeywordStatisticsList]
+    ) -> Tuple[int, int]:
         """
         Adds statistics from provided list to app.
         :param statistics: List of KeywordStatistics objects
         :return: list of dictionaries with collection name and number of keywords.
         """
         collections, keywords = set(), set()
-        for stat in statistics:
-            stat_req = self.client.add_statistics(KeywordStatisticsList.one(stat))
-            if stat_req[0] == 201:
-                collections.add(stat.collection)
-                keywords.add(".".join((stat.collection, stat.keyword)))
-            elif stat_req[0] != 400:
-                print(stat_req[1]["detail"])
+        progress_bar = IncrementalBar(
+            "Sending statistics",
+            max=len(statistics),
+            suffix="%(percent).1f%% - %(eta)ds, elapsed: %(elapsed)ds",
+        )
+        for stats in statistics:
+            stats_req = self.client.add_statistics(KeywordStatisticsList.of(stats))
+            if stats_req[0] == 201:
+                collections.update({stat.collection for stat in stats})
+                keywords.update(
+                    {".".join((stat.collection, stat.keyword)) for stat in stats}
+                )
+                progress_bar.next()
+            elif stats_req[0] != 400:
+                print(stats_req[1]["detail"])
                 raise StopIteration
             else:
                 print(
-                    f"""Record already exists for provided collection: {
-                    stat.collection}, keyword: {stat.keyword} and execution_time: {stat.execution_time}"""
+                    f"""Record already exists for provided execution_time: {stats[0].execution_time}"""
                 )
-
         return len(collections), len(keywords)
```

Comment:
Do you think that this return information is required? The method is uploading statistics records for multiple files and we are not showing anywhere how much statistics records we uploaded, but only the number of unique collections and keywords across all the files. It seems like a secondary information. Maybe you could also present the number of statistics uploaded? If you upload N files with different executions of the same suite, the number of collections and keywords may not reflect the number of statistics uploaded which can be up to N*keywords

Reply:
Yes, You are totally right. I could change it according to your comment. I wanted to keep both importers to work the same way and return `Tuple[int, int]` that is used [here](https://github.com/pbylicki/rfhub2/blob/master/rfhub2/cli/cli.py#L64). But still, I can add total number of collections and keywords, instead of unique ones. Is that alright?

Reply:
I think it would be good to at least log the total number of statistic records uploaded. If you want to keep the numbers of collections and keywords they should be unique to not produce misleading message. After all you are counting here references to collections and keywords, not these objects itself

Reply:
I will change the message to files and statistics number, that would be clearer.

Reply:
Good idea!